benchmark/bench/codspeed.js

```diff
@@ -0,0 +1,107 @@
+import fs from 'node:fs';
+import { fileURLToPath } from 'node:url';
+import { withCodSpeed } from '@codspeed/tinybench-plugin';
+import { waitUntilBusy } from 'port-authority';
+import { Bench } from 'tinybench';
+import { exec } from 'tinyexec';
+import { astroBin } from './_util.js';
+import { benchmarkRenderTime } from './render.js';
+import { benchmarkCannon } from './server-stress.js';
+
+export async function run({ memory, render, _stress }) {
+	const bench = process.env.CODSPEED ? withCodSpeed(new Bench()) : new Bench();
+
+	bench
+		.add('memory', async () => {
+			await exec('node', ['--expose-gc', '--max_old_space_size=10000', astroBin, 'build'], {
+				nodeOptions: {
+					cwd: memory.root,
+					stdio: 'inherit',
+					env: {
+						ASTRO_TIMER_PATH: memory.output,
+					},
+				},
+			});
+		})
+		.add(
+			'render',
+			async () => {
+				console.info('Bench rendering.');
+				const port = 4322;
+
+				console.info('Previewing...');
+				const previewProcess = exec(astroBin, ['preview', '--port', port], {
+					nodeOptions: {
+						cwd: render.root,
+						stdio: 'inherit',
+					},
+				});
```

Comment:
From the CI logs, looks like this is still failing somehow. The console should log:

```
Preview server listening on http://localhost:4322
```

Maybe it's easier to skip this test for now, or to debug it you can pass `throwOnError: true` besides the `nodeOptions` option. 

I need to followup and improve the error logging for tinyexec because it doesn't throw for exit code 1 by default 😢